Validate owners with a file / GitHub API (#59)
toptal/codeowners-checker: README.md

It will configure `@owner` as the default owner in the config file.

### Generating and validating owners

By default, owners will be validated either against prepopulated file (teams.txt) or against data fetched from github.
Format of teams.txt is one owner per line.

> Example teams.txt:

Reviewer: I know I suggested the file name, now that I think of it `owners.txt` sounds better.

```
@company/backend-devs
@company/frontend-devs
@john.smith
```
Github credentials are taken from the following environment variables. You might want to put them into your .bashrc (or equivalent):

>     $ export GITHUB_TOKEN='your github PAT' # your personal access token from github

Reviewer: Could you add a link pointing out to how to get the token.

>     $ export GITHUB_organization='company' # name of your github organization

Reviewer 2: ```suggestion
    $ export GITHUB_ORGANIZATION='company' # name of your github organization
```

>     $ export GITHUB_organization='company' # name of your github organization

Reviewer: Could we parse it from `git config --get remote.origin.url`? We could make the variable optional.

Also maybe `codeowners-checker config organization` like we do with `owner`?

BTW it's a good habit to use all upper case for global shell variables.

It will configure `@owner` as the default owner in the config file.

### Fetching and validating owners

> By default, [check](#check-file-consistency) command wil validate owners either against prepopulated file (OWNERS)

Reviewer 2: ```suggestion
By default, [check](#check-file-consistency) command will validate owners either against the prepopulated file (OWNERS)
```

> or against data fetched from github.

Reviewer 2: ```suggestion
or against data fetched from Github.
```

Format of OWNERS is one owner per line.

Example OWNERS:
```
@company/backend-devs
@company/frontend-devs
@john.smith
```
Github credentials are taken from the following environment variables. You might want to put them into your .bashrc or equivalent:

    $ export GITHUB_TOKEN='your github PAT' # your personal access token from github
    $ export GITHUB_ORGANIZATION='company' # name of your github organization

> You can generate your PAT in [Settings -> Developer settings -> Personal access tokens on GitHub](https://github.com/settings/tokens) and `read:org` scope is ***required***.

Reviewer 2: two stars are enough on **required**
```suggestion
You can generate your PAT in [Settings -> Developer settings -> Personal access tokens on GitHub](https://github.com/settings/tokens) and `read:org` scope is **required**.
```
